scripts/nb-tester/qiskit_docs_notebook_tester/__init__.py

```diff
@@ -261,7 +305,29 @@ async def _execute_notebook(filepath: Path, config: Config) -> nbformat.Notebook
 
     await _execute_in_kernel(kernel, PRE_EXECUTE_CODE)
     if config.should_patch(filepath):
-        await _execute_in_kernel(kernel, MOCKING_CODE)
+        def get_arg(arg, default):
+            return vars(config.args).get(arg.lstrip("-").replace("-", "_"), default)
```

Comment:
This shouldn't be necessary? argparse will already handle for you converting `-` into `_` and applying defaults. It's a code smell that you have to specify the default twice atm.

What was the issue with usin `config.args.backend` et al?